caillef/dustzh.lua

```diff
@@ -967,14 +1021,30 @@ weaponsMetatable = {
                 self.currentWeapon = weaponInfo
                 self.maxAmmo = weaponInfo.ammo
                 self.ammo = weaponInfo.ammo
+                self.weaponName = weaponInfo.name
                 self:updateAmmoUI()
+                self:updateNameUI()
                 multi:playerAction("changeWeapon", { id = id })
             end
 			p.weaponId = id
 			if self.templates[weaponInfo.item] then
 				local weapon =  Shape(self.templates[weaponInfo.item])
 				weapon.Pivot = self.templates[weaponInfo.item].Pivot
 				self:_setWeapon(p, weapon, weaponInfo, forceNotFPS)
+
+                -- display the weapon next to the weapon name
+                for _, i in ipairs(displayedWeapon) do
+                    Camera:RemoveChild(i)
+                end
+
+                displayedWeapon = {}
+                ui = require("uikit")
+                local displayedWeapon = ui:createShape(Shape(self.templates[weaponInfo.item]))
+                displayedWeapon.parentDidResize = function()
```

Comment:
it looks like you should also update the size of the displayed weapon, like you do for the weapon's name.
Currently it always has the same height and width.

Small screen:
<img width="261" alt="Screenshot 2023-07-31 at 10 43 24" src="https://github.com/cubzh/worlds/assets/939999/1d13debd-f388-4522-90dd-0e92c26669e9">

Large screen:
<img width="237" alt="Screenshot 2023-07-31 at 10 43 31" src="https://github.com/cubzh/worlds/assets/939999/91331bea-e836-496a-8618-98a23580d8a1">